### ISSUE TITLE
Picture taken with camera does not contain location data but gallery images do

### ISSUE THREAD

**Issue by Thomasssb1** [flutter/flutter#121628]
I have got an app which includes image_picker to allow the user to choose an image from their gallery or take a photo directly within the app. I need the photo to contain location data so I need the **GPS GPSLatitude** tag to be inside of the metadata of the image.
When taking a photo just before inputting it through the gallery, the location data is saved to the metadata - whereas taking a photo in app does not contain location despite the setting save location being enabled on my devices camera settings.

Essentially, the code I am using is very simple; using the [exif ](https://pub.dev/packages/exif) library to get metadata along with image_picker. Code is below:
```dart
var image = await picker.pickImage(source: ImageSource.camera);
var metadata = await readExifFromBytes(File(image.path).readAsBytesSync());
// check for gps tag
if (metadata.containsKey('GPS GPSLatitude')) {
      print("location enabled");
else{
      print("location disabled");
};
```
The device I am using is a Xiaomi Redmi Note 8T running on android version 10.

**Comment by darshankawar** [comment]
@Thomasssb1 
I think your case seems to be similar to https://github.com/flutter/flutter/issues/74615. Please check and confirm. If so, we can follow-up in that issue for better tracking.

**Comment by darshankawar** [comment]
@Thomasssb1 
Coming back to the issue, my colleague verified this behavior and observed that the functionality is working as expected:

You need to request for permission in the app before you can have access to the location data.
Using below code sample, the location data is received properly:

<details>
<summary> code sample </summary>

```

import 'dart:io';

import 'package:flutter/material.dart';
import 'package:image_picker/image_picker.dart';
import 'package:exif/exif.dart';
import 'package:location/location.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData.dark(
        useMaterial3: true,
      ),
      home: Home(),
    );
  }
}

class Home extends StatefulWidget {
  const Home({
    super.key,
  });

  @override
  State<Home> createState() => _HomeState();
}

class _HomeState extends State<Home> {
  File? pickedImage;
  Map<String, Object?>? _exif;

  @override
  void initState() {
    super.initState();

    requestPermission();
  }

  requestPermission() async {
    Location location = Location();

    bool _serviceEnabled = await location.serviceEnabled();
    if (!_serviceEnabled) {
      _serviceEnabled = await location.requestService();
      if (!_serviceEnabled) {
        return;
      }
    }

    PermissionStatus _permissionGranted;
    _permissionGranted = await location.hasPermission();
    if (_permissionGranted == PermissionStatus.denied) {
      _permissionGranted = await location.requestPermission();
      if (_permissionGranted != PermissionStatus.granted) {
        return;
      }
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: ListView(
          children: [
            if (pickedImage != null) Image.file(pickedImage!),
            const SizedBox(
              height: 20,
            ),
            if (_exif != null) Text(_exif.toString()),
            const SizedBox(
              height: 20,
            ),
            FilledButton(
              style: ButtonStyle(
                fixedSize: MaterialStateProperty.all(const Size(80, 80)),
                shape: MaterialStateProperty.resolveWith(
                  (states) => RoundedRectangleBorder(
                    borderRadius: BorderRadius.circular(
                      states.contains(MaterialState.pressed) ? 20 : 50,
                    ),
                  ),
                ),
              ),
              child: const Text('Source Camera'),
              onPressed: () async {
                final picker = ImagePicker();
                final image = await picker.pickImage(source: ImageSource.camera, requestFullMetadata: true);
                if (image != null) {
                  final file = File(image.path);
                  final exif = await readExifFromBytes(file.readAsBytesSync());
                  _exif = exif;
                  setState(() {
                    pickedImage = file;
                  });
                }
              },
            ),
            FilledButton(
              style: ButtonStyle(
                fixedSize: MaterialStateProperty.all(const Size(80, 80)),
                shape: MaterialStateProperty.resolveWith(
                  (states) => RoundedRectangleBorder(
                    borderRadius: BorderRadius.circular(
                      states.contains(MaterialState.pressed) ? 20 : 50,
                    ),
                  ),
                ),
              ),
              child: const Text('Source Gallery'),
              onPressed: () async {
                final picker = ImagePicker();
                final image = await picker.pickImage(source: ImageSource.gallery, requestFullMetadata: true);
                if (image != null) {
                  final file = File(image.path);
                  final exif = await readExifFromBytes(file.readAsBytesSync());
                  _exif = exif;
                  setState(() {
                    pickedImage = file;
                  });
                }
              },
            ),
          ],
        ),
      ),
    );
  }
}

```

</details>

You can try the same and verify at your end. Since it works as expected, I am going ahead and closing it as not an issue. If you disagree, write in comments and I'll reopen it.

**Comment by Thomasssb1** [comment]
This did not fix the problem, I already had the location permission enabled - but yet when reading from camera it would not return location data.

I tested it again and to my surprise it worked - without any changes. However, it did not work when I took photos in certain areas of my house; but worked in other areas. I have seen something a while ago about needing to have visible sky in an image in order for location data to be embedded within the image, but never thought that it was true - however it seemed to be working whenever I took photos of the sky in it or just randomly would work. 
Would there be a way to override this possible setting? (I don't really know if it is true) 
[reference to sky comment](https://github.com/flutter/flutter/issues/99817#issuecomment-1064875457)
